### ISSUE TITLE
Unsafe overlap checks for runtime checkable protocols don't consider inheritance

### ISSUE THREAD

**Issue by fangyi-zhou** [microsoft/pyright#11131]
**Describe the bug**

See snippet below. An error is reported for checking whether `No()` is of instance `UnsafeProtocol` but not of instance `ChildUnsafeProtocol`.

**Code or Screenshots**

```python
from typing import Protocol, runtime_checkable

@runtime_checkable
class UnsafeProtocol(Protocol):
    def foo(self) -> int: ...

@runtime_checkable
class ChildUnsafeProtocol(UnsafeProtocol):
    def bar(self) -> str: ...

class No:
    def foo(self) -> str:
        return "not an int"
    def bar(self) -> int:
        return 42

isinstance(No(), UnsafeProtocol)
isinstance(No(), ChildUnsafeProtocol)  # No error
```

https://pyright-play.net/?strict=true&analyzeUnannotatedFunctions=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYA0UIArijFgKYD6pAFs6QNYCGAIwrMAUCIAC9Ri3Zdeg4SNIU%2BAZ1VQAqilV9gzImBLkKACkPHKASgBcIqA6gATZsCjAwYU6uYVgVqABaAD5MRhsoADposUkGJgg2Tm5%2BIVFlNQ0AYQ4kCidtXX0LMkpTQr0DYlKKW3tHFzcBPhBvX38g0NUYEAjoyLEM9SgAOTA7R2dXd082vwCQqG7e%2BsmHEGYYOhAUKAAiFCMoPl3UGD3VqaaWuY7Fs4m19c3t3YAWACYxJFVUbpPSMxTGNTFYaBVitUTFYRD8-jAAUCQWCoDk8gUdJUStCHABiUZgKDMEDgEBAA

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?
